resources/main-template.cc

```diff
@@ -50,9 +51,54 @@ void TearDownOncePerProcess();
 }
 #endif
 namespace boxednode {
+namespace {
+struct TimingEntry {
+  const char* const label;
+  uint64_t const time;
+  TimingEntry* next = nullptr;
+  ~TimingEntry() {
+    delete next;
+  }
+};
+TimingEntry start_time_entry { "Process initialization", uv_hrtime() };
+std::atomic<TimingEntry*> current_time_entry { &start_time_entry };
```

Comment:
nit: do we need it to be atomic? Is the startup process multithread?

Reply:
It's not, but I'd like to avoid a situation where somebody in the future adds a `MarkTime()` call that does end up running in another thread (it's supposed to be a simple, straightforward thing to add) and then crashes the process ;)